src/TMGToolbox/inputOutput/importMatrixFromCSVThirdNormalized.py

```diff
@@ -15,149 +15,273 @@
+    startTime = time(int(initialTime[0]),int(initialTime[1]),int(initialTime[2]),int(initialTime[3]))
+    matrix.setFrom(startTime)
+    durationTime = durationTime.split(":")
+    matrix.setDuration(GKTimeDuration(int(durationTime[0]), int(durationTime[1]), int(durationTime[2])))
+    
+    #read file and import
+    with open(fileLocation) as csvfile:
+        reader = csv.reader(csvfile)
+        sectionType = model.getType("GKCentroid")
+        entranceCentroidType = model.getType("GKPedestrianEntranceCentroid")
+        exitCentroidType = model.getType("GKPedestrianExitCentroid")
+
+        if header is True:
+            next(reader)
+        for line in reader:
+            if thirdNormalized is True:
+                value = float(line[2])
+                # Only create object if OD value is non zero
+                if value != 0.0:
+                    if vehicleEID == 'transit':
+                        originEID = f"ped_entrance_centroid_{line[0]}"
+                        destinationEID = f"ped_exit_centroid_{line[1]}"
+                        origin = catalog.findObjectByExternalId(originEID, entranceCentroidType)
+                        destination = catalog.findObjectByExternalId(destinationEID, exitCentroidType)
+                    else:
+                        originEID = f"centroid_{line[0]}"
+                        destinationEID = f"centroid_{line[1]}"
+                        origin = catalog.findObjectByExternalId(originEID, sectionType)
+                        destination = catalog.findObjectByExternalId(destinationEID, sectionType)
+                    if origin is None:
+                        raise Exception(f"The specified centroid '{originEID}' does not exist")
+                    if destination is None:
+                        raise Exception(f"The specified centroid '{destinationEID}' does not exist")
+                    matrix.setTrips(origin, destination, value)
+            else:
+                raise Exception("Functionality has not been implemented yet")
 
-#read file and import
+    # Save add the matrix to the network file
+    folderName = "GKCentroidConfiguration::matrices"
+    folder = model.getCreateRootFolder().findFolder( folderName )
+    if folder is None:
+        folder = GKSystem.getSystem().createFolder( model.getCreateRootFolder(), folderName )
+    folder.append(matrix)
+    # Save the network file
+    print("Save Network")
+    console.save(outputNetworkFile)
 
-with open(fileLocation) as csvfile:
-    reader = csv.reader(csvfile)
-    sectionType = model.getType("GKCentroid")
-    entranceCentroidType = model.getType("GKPedestrianEntranceCentroid")
-    exitCentroidType = model.getType("GKPedestrianExitCentroid")
+def runFromConsole(argv):
+    """
+    this function is used to run the script from the terminal. Since it 
+    needs a csv to pass in the parameters making it to run from _execute() 
+    is too difficult and so we will leave this feature as is untouched.
```

Comment:
I am sure that you could reconstruct the parameters and pass it into _execute instead of doubling the code.

Reply:
Fixed and resolved